common/sgx_workload/workload/ext_work_order_info_impl.h

```diff
@@ -0,0 +1,70 @@
+/* Copyright 2020 Intel Corporation
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*     http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+#include <stdlib.h>
+
+#include "types.h"
+#include "ext_work_order_info.h"
+
+// ExtWorkOrderInfoImpl class implements ExtWorkOrderInfo interface
+class ExtWorkOrderInfoImpl : public ExtWorkOrderInfo {
```

Comment:
class ExtWorkOrderInfo being extended in 3 classes. Are all those classes has to implementation of pure virtual functions differently? If they use common logic then can we use abstract class with default implementations?

Reply:
There are common implementation of few APIs which needs to be accessed from KME and WPE. So, i extended implementation of ExtWorkOrderInfo (abstract class) into KME and WPE.